databricks_cli/runs/cli.py

```diff
@@ -106,12 +112,32 @@ def list_cli(api_client, job_id, active_only, completed_only, offset, limit, out
       - Result state (can be n/a)
     """
     check_version(api_client, version)
-    runs_json = RunsApi(api_client).list_runs(
-        job_id, active_only, completed_only, offset, limit, version=version)
+    api_version = version or api_client.jobs_api_version
+    if api_version != '2.1' and (expand_tasks or offset or limit or _all):
+        click.echo(click.style('ERROR', fg='red') + ': the options --expand-tasks, ' +
+                   '--offset, --limit, and --all are only available in API 2.1', err=True)
+        return
+    runs_api = RunsApi(api_client)
+
+    has_more = True
+    runs = []
+    if _all:
+        offset = 0
+        limit = 20
+    while has_more:
+        runs_json = runs_api.list_runs(
+            job_id, active_only, completed_only, offset, limit,
+            version=version, expand_tasks=expand_tasks)
+        runs += runs_json['runs'] if 'runs' in runs_json else []
+        has_more = runs_json.get('has_more', False) and _all
+        if has_more:
+            offset = offset + \
+                (len(runs_json['runs']) if 'runs' in runs_json else limit)
```

Comment:
@nordp The command itself has offset and limit arguments. I think you'd want to make `--all` mutually exclusive with specifying `--offset` and `--limit`, and then introduce a separate branch for `--all` here to put the logic for repeated calls to the API. Alternatively, right after L131, something like:

```py
if not _all:
  break
else:
  has_more = runs_json.get('has_more', False)
  # ...
```

Reply:
I don't think I want to make it mutually exclusive, because a user might want to ask for a higher limit than what is supported by the API, but still not fetch ALL jobs/runs.